CRM/Financial/BAO/Order.php

```diff
@@ -686,7 +686,7 @@ protected function setPriceFieldMetadata(array $metadata): void {
             if ($membershipType['auto_renew'] && empty($this->priceSetMetadata['auto_renew_membership_field'])) {
               // Quick form layer supports one auto-renew membership type per price set. If we
               // want more for any reason we can add another array property.
-              $this->priceSetMetadata['auto_renew_membership_field'] = (int) $option['membership_type_id'];
+              $this->priceSetMetadata['auto_renew_membership_field'] = (int) $option['id'];
```

Comment:
If I am right in my other comment then I think this needs to be $priceField['id'] right as per https://github.com/civicrm/civicrm-core/blob/c4b3726f5bc4abd69c37a7b4ddcb5e352fa52e34/CRM/Price/BAO/PriceSet.php#L846